api.py

```diff
@@ -433,8 +433,8 @@ def get_terms_via_payload(collection: Collection, payload: Query, field: TermFie
     return _get_terms(collection, payload.q, field.value, aggr.value)
 
 
-@v1.get("/{collection}/article/{id}", tags=["data"])
-@v1.head("/{collection}/article/{id}", include_in_schema=False)
+@v1.get("/{collection}/article/{id:path}", tags=["data"])
+@v1.head("/{collection}/article/{id:path}", include_in_schema=False)
```

Comment:
This will not work when the original URI (and its SURT) has query parameters. In a prior version of the code I used base64-encoded ID, but removed it when we decided to go with a hash digest-based ID. Unfortunately, some records are still using the plain SURT as IDs, so we need to go back to the base64-encoding approach.

Reply:
Suggestion:
In `Flask/Werkzeug` we use `request.environ['RAW_URI']` to get the full URL of the request.
There should be something similar if FastAPI.

The raw URL would be `/mediacloud/article/<target-urlkey>`, then you'll strip the `/mediacloud/article/` and you'll have the user's request.